src/N3Writer.js

```diff
@@ -58,9 +58,30 @@ export default class N3Writer {
       this._prefixIRIs = Object.create(null);
       options.prefixes && this.addPrefixes(options.prefixes);
       if (options.baseIRI) {
-        this._baseMatcher = new RegExp(`^${escapeRegex(options.baseIRI)
-            }${options.baseIRI.endsWith('/') ? '' : '[#?]'}`);
+        const escapedBaseIRI = escapeRegex(options.baseIRI) + (options.baseIRI.endsWith('/') ? '' : '[#?]');
+        const escapedSchemeDelimiter = escapeRegex('://');
```

Comment:
I can't think of an edge case right away, but I would do this in the raw domain rather than the escaped domain.